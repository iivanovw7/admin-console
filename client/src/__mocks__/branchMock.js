export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};