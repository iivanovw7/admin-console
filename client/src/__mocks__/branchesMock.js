export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
