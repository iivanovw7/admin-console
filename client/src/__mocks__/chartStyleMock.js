export const style = {
  type: "PieChart",
  id: 2
};
