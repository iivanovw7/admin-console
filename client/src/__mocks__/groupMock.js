export const GROUP = {
  status: true,
  permissions: true,
  _id: '5c8f96bacece9911f01dd0a6',
  name: 'Front-End Pool',
  description: 'Group of all employees with Front-End expertise'
};