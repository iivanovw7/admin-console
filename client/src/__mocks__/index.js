export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';
export * from './groupMock';
export * from './groupsMock';
export * from './roleMock';
export * from './rolesMock';
export * from './ticketMock';
export * from './ticketsMock';
export * from './messageMocks';
export * from './messagesMocks';
export * from './statisticsMock';
