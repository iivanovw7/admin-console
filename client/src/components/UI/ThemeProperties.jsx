import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

