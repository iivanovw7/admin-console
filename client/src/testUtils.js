import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};