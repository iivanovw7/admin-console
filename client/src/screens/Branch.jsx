import { Paper } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import BranchContainer from '../components/Branch/BranchContainer';
import { Wrapper } from '../components/UI/ThemeProperties';

const Branch = props => {
  const { classes, history, dispatch } = props;
  const branch = props.branches.branch;

  return (
    <main className={classes.contentSingle}>
      <Paper className={classes.titleContainer}>
        <div className={classes.selectorsContainer}>
          <h2>Create/Edit branch</h2>
        </div>
      </Paper>
      <BranchContainer branch={branch} history={history} dispatch={dispatch}/>
    </main>
  );
};

Branch.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    branches: state.branches
  };
}

export default connect(mapStateToProps)(withStyles(Wrapper, { withTheme: true })(withRouter(Branch)));
