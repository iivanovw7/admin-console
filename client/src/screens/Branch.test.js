import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import jest from 'jest-mock';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import * as mocks from '../__mocks__/';
import { Wrapper } from '../components/UI/ThemeProperties';
import Branch from '../screens/Branch';
import { mockStore } from '../testUtils';

const setup = (initialState = mockStore({})) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  const mockFunc = jest.fn();
  const Composition = withStyles(Wrapper, { withTheme: true })(withRouter(Branch));
  const props = {
    dispatch: mockFunc,
    branches: {
      branch: mocks.BRANCH
    }
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders branch main screen form with styles,
   *  Checks if component rendered properly.
   */
  describe('Branch screen', () => {
    it('Should match snapshot', () => {
      wrapper = shallow(
        <Provider store={initialState}>
          <MemoryRouter>
            <Composition {...props} dispatch={mockFunc}/>
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.length).toEqual(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
};

setup();



