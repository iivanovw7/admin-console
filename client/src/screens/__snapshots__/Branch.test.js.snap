// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branch screen Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <MemoryRouter>
      <WithStyles(withRouter(Connect(WithStyles(withRouter(Branch)))))
        branches={
          Object {
            "branch": Object {
              "__v": 0,
              "_id": "5c8f96b9cece9911f01dd092",
              "address": "938 Barby Way",
              "email": "mbrazier0@amazon.co.uk",
              "fax": "+86 829 425 7497",
              "information": "1",
              "name": "San Martín de los Andes",
              "phone": "+54 275 474 9318",
              "status": true,
            },
          }
        }
        dispatch={[MockFunction]}
      />
    </MemoryRouter>
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <MemoryRouter>
        <WithStyles(withRouter(Connect(WithStyles(withRouter(Branch)))))
          branches={
            Object {
              "branch": Object {
                "__v": 0,
                "_id": "5c8f96b9cece9911f01dd092",
                "address": "938 Barby Way",
                "email": "mbrazier0@amazon.co.uk",
                "fax": "+86 829 425 7497",
                "information": "1",
                "name": "San Martín de los Andes",
                "phone": "+54 275 474 9318",
                "status": true,
              },
            }
          }
          dispatch={[MockFunction]}
        />
      </MemoryRouter>,
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeState": Object {},
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(withRouter(Connect(WithStyles(withRouter(Branch)))))
          branches={
            Object {
              "branch": Object {
                "__v": 0,
                "_id": "5c8f96b9cece9911f01dd092",
                "address": "938 Barby Way",
                "email": "mbrazier0@amazon.co.uk",
                "fax": "+86 829 425 7497",
                "information": "1",
                "name": "San Martín de los Andes",
                "phone": "+54 275 474 9318",
                "status": true,
              },
            }
          }
          dispatch={[MockFunction]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "branches": Object {
            "branch": Object {
              "__v": 0,
              "_id": "5c8f96b9cece9911f01dd092",
              "address": "938 Barby Way",
              "email": "mbrazier0@amazon.co.uk",
              "fax": "+86 829 425 7497",
              "information": "1",
              "name": "San Martín de los Andes",
              "phone": "+54 275 474 9318",
              "status": true,
            },
          },
          "dispatch": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <MemoryRouter>
          <WithStyles(withRouter(Connect(WithStyles(withRouter(Branch)))))
            branches={
              Object {
                "branch": Object {
                  "__v": 0,
                  "_id": "5c8f96b9cece9911f01dd092",
                  "address": "938 Barby Way",
                  "email": "mbrazier0@amazon.co.uk",
                  "fax": "+86 829 425 7497",
                  "information": "1",
                  "name": "San Martín de los Andes",
                  "phone": "+54 275 474 9318",
                  "status": true,
                },
              }
            }
            dispatch={[MockFunction]}
          />
        </MemoryRouter>,
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "storeState": Object {},
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(withRouter(Connect(WithStyles(withRouter(Branch)))))
            branches={
              Object {
                "branch": Object {
                  "__v": 0,
                  "_id": "5c8f96b9cece9911f01dd092",
                  "address": "938 Barby Way",
                  "email": "mbrazier0@amazon.co.uk",
                  "fax": "+86 829 425 7497",
                  "information": "1",
                  "name": "San Martín de los Andes",
                  "phone": "+54 275 474 9318",
                  "status": true,
                },
              }
            }
            dispatch={[MockFunction]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "branches": Object {
              "branch": Object {
                "__v": 0,
                "_id": "5c8f96b9cece9911f01dd092",
                "address": "938 Barby Way",
                "email": "mbrazier0@amazon.co.uk",
                "fax": "+86 829 425 7497",
                "information": "1",
                "name": "San Martín de los Andes",
                "phone": "+54 275 474 9318",
                "status": true,
              },
            },
            "dispatch": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
  Symbol(enzyme.__childContext__): null,
}
`;
